lib/pbench/server/api/resources/query_apis/metadata_index/__init__.py

```diff
@@ -0,0 +1,103 @@
+
+
+class RunIdBase(ElasticBase):
+    """
+    A base class for query apis that depends on Metadata for getting the
+    indices.
+
+    This class extends the ElasticBase class and implements a common
+    `preprocess` method based on client provided run id. The subsequent methods
+    such as 'assemble' and 'postprocess' need to be implemented by a respective
+    subclasses.
+
+    Note that run_id is a required schema parameter for all the
+    classes inheriting this class. Also "preprocess" provides json context
+    containing dataset and a run_id that's passed to the assemble and
+    postprocess methods.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger, schema: Schema):
+        if "run_id" not in schema:
+            raise MissingRunIdSchemaParameter(self.__class__.__name__)
+        super().__init__(config, logger, schema)
+
+    def preprocess(self, client_json: JSON) -> CONTEXT:
+        """
+        Query the Dataset associated with this run id, and determine whether the
+        request is authorized for this dataset.
+
+        If the user has authorization to read the dataset, return the Dataset
+        object and run_id as JSON CONTEXT so that the postprocess operations
+        can use it to identify the index to be searched from document index
+        metadata.
+        """
+        run_id = client_json["run_id"]
+
+        # Query the dataset using the given run id
+        dataset = Dataset.query(md5=run_id)
+        if not dataset:
+            self.logger.error(f"Dataset with Run ID {run_id!r} not found")
+            abort(HTTPStatus.NOT_FOUND, message="Dataset not found")
+
+        owner = User.query(id=dataset.owner_id)
+        if not owner:
+            self.logger.error(
+                f"Dataset owner ID { dataset.owner_id!r} cannot be found in Users"
+            )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="Dataset owner not found")
+
+        # We check authorization against the ownership of the dataset that
+        # was selected rather than having an explicit "user"
+        # JSON parameter. This will raise UnauthorizedAccess on failure.
+        self._check_authorization(owner.username, dataset.access)
+
+        # The dataset exists, and authenticated user has enough access so continue
+        # the operation with the appropriate CONTEXT.
+        return {"dataset": dataset, "run_id": run_id}
+
+    def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
+        """
+        Retrieve the list of ES indices from the metadata table based on a given
+        root_index_name.
+        """
+        try:
+            index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
+            index_keys = [key for key in index_map if root_index_name in key]
+        except MetadataError as e:
+            self.logger.error(f"Indices from metadata table not found {e!r}")
```

Comment:
Like Dave said, not a showstopper, but if you happen make other changes here, it would probably be good to move the assignment of `index_keys` out of the `try` block, since any exceptions that _it_ raises won't be caught here.  (It's a good idea for the `try` block to cover as small a range of code as reasonable and not to have extraneous statements inside it.)

Reply:
It seems like there are no outstanding conversations that needs addressing so if this is not a showstopper should I squash all the commits?

Reply:
Ah I still need to address this one https://github.com/distributed-system-analysis/pbench/pull/2440#discussion_r741865343 NM

Reply:
> It seems like there are no outstanding conversations that needs addressing so if this is not a showstopper should I squash all the commits?

I see 6 open conversations. A few of them were already addressed, but some aren't. The main open issue for me is the current lack of "unit" testing of the no-filter case.

Reply:
This wasn't wasn't technically resolved; but since I started it, and both Webb and I called it "not a showstopper", I'm going to mark it resolved. (I still wouldn't mind if you fixed it, should you end up needing to touch this file again.)